source/javascripts/lib/wirsing/filters/search.coffee

```diff
@@ -1,24 +1,37 @@
 #= require ../module
 
-# Returns a configurable search function that can be used
-# to filter an array of objects by searching for a term
-# within the specified property of each object.
-
 wirsing.module 'filter.search', ->
 
+  # Returns a configurable search function that can be used
+  # to filter an array of objects by searching for a term
+  # within the specified property of each object.
+
   (options) ->
 
+    # This is the function that is returned to the outside world.
+    # It takes an array of objects and returns an array of matching objects
     search = (items) ->
       if search.term.length < search.minChars
         items # dont filter
       else
-        items.filter (item) ->
-          item[search.in].toLowerCase().search(search.term.toLowerCase()) > -1
+        items.filter (item) -> search.in.some (key) ->
```

Comment:
its a standard method on arrays that loops through all elements (lik `filter`) but it returns `true` if the test function returns `true` for some elements (at least one).